enterprise/suppress-user/factory_test.go

```diff
@@ -118,3 +125,60 @@ func getSuppressionFromManager(w http.ResponseWriter, _ *http.Request) {
 	}
 	_, _ = w.Write(body)
 }
+
+func generateTestData(t *testing.T, tmpDir string) {
+	repo, err := badgerdb.NewRepository(tmpDir, logger.NOP, stats.Default)
+	require.NoError(t, err)
+
+	suppressions := []model.Suppression{
+		{
+			Canceled:    false,
+			WorkspaceID: "workspace-1",
+			UserID:      "user-1",
+			SourceIDs:   []string{"src-1"},
+		},
+		{
+			Canceled:    false,
+			WorkspaceID: "workspace-1",
+			UserID:      "user-1",
+			SourceIDs:   []string{"src-2"},
+		},
+	}
+	generateBackupFiles(t, tmpDir, "latest-export", repo, suppressions)
+
+	suppressions = append(suppressions, []model.Suppression{
+		{
+			Canceled:    false,
+			WorkspaceID: "workspace-2",
+			UserID:      "user-2",
+			SourceIDs:   []string{"src-3"},
+		},
+		{
+			Canceled:    false,
+			WorkspaceID: "workspace-2",
+			UserID:      "user-2",
+			SourceIDs:   []string{"src-4"},
+		},
+	}...)
+	generateBackupFiles(t, tmpDir, "full-export", repo, suppressions)
+}
+
+func generateBackupFiles(t *testing.T, dir, backupFilename string, repo *badgerdb.Repository, suppressions []model.Suppression) {
+	token := []byte("__token__")
+
+	repo.Add(suppressions, token)
+
+	f, err := os.Create(path.Join(dir, backupFilename))
+	defer func() { _ = f.Close() }()
+	require.NoError(t, err)
+	require.NoError(t, repo.Backup(f))
+}
```

Comment:
can this be converted into a function local to `generateTestData`?

Reply:
Done